core/ir/arm/encode.c

```diff
@@ -461,6 +461,238 @@ const reg_id_t dr_reg_fixer[] = {
+    DR_REG_V35, /* DR_REG_Q19 */
+    DR_REG_V36, /* DR_REG_Q20 */
+    DR_REG_V37, /* DR_REG_Q21 */
+    DR_REG_V38, /* DR_REG_Q22 */
+    DR_REG_V39, /* DR_REG_Q23 */
+    DR_REG_V40, /* DR_REG_Q24 */
+    DR_REG_V41, /* DR_REG_Q25 */
+    DR_REG_V42, /* DR_REG_Q26 */
+    DR_REG_V43, /* DR_REG_Q27 */
+    DR_REG_V44, /* DR_REG_Q28 */
+    DR_REG_V45, /* DR_REG_Q29 */
+    DR_REG_V46, /* DR_REG_Q30 */
+    DR_REG_V47, /* DR_REG_Q31 */
+
+    /* For AArch64, the smaller SIMD names refer to the lower
+     * bits of the corresponding same-number larger SIMD register.
+     * But for AArch32, the smaller ones are compressed such that
+     * they refer to the top and bottom.  B and H are AArch64-only.
+     */
+    /* from DR_REG_D0 to DR_REG_D31 */
+    DR_REG_V16, /* DR_REG_D0 */
+    DR_REG_V16, /* DR_REG_D1 */
+    DR_REG_V17, /* DR_REG_D2 */
+    DR_REG_V17, /* DR_REG_D3 */
+    DR_REG_V18, /* DR_REG_D4 */
+    DR_REG_V18, /* DR_REG_D5 */
+    DR_REG_V19, /* DR_REG_D6 */
+    DR_REG_V19, /* DR_REG_D7 */
+    DR_REG_V20, /* DR_REG_D8 */
+    DR_REG_V20, /* DR_REG_D9 */
+    DR_REG_V21, /* DR_REG_D10 */
+    DR_REG_V21, /* DR_REG_D11 */
+    DR_REG_V22, /* DR_REG_D12 */
+    DR_REG_V22, /* DR_REG_D13 */
+    DR_REG_V23, /* DR_REG_D14 */
+    DR_REG_V23, /* DR_REG_D15 */
+    DR_REG_V24, /* DR_REG_D16 */
+    DR_REG_V24, /* DR_REG_D17 */
+    DR_REG_V25, /* DR_REG_D18 */
+    DR_REG_V25, /* DR_REG_D19 */
+    DR_REG_V26, /* DR_REG_D20 */
+    DR_REG_V26, /* DR_REG_D21 */
+    DR_REG_V27, /* DR_REG_D22 */
+    DR_REG_V27, /* DR_REG_D23 */
+    DR_REG_V28, /* DR_REG_D24 */
+    DR_REG_V28, /* DR_REG_D25 */
+    DR_REG_V29, /* DR_REG_D26 */
+    DR_REG_V29, /* DR_REG_D27 */
+    DR_REG_V30, /* DR_REG_D28 */
+    DR_REG_V30, /* DR_REG_D29 */
+    DR_REG_V31, /* DR_REG_D30 */
+    DR_REG_V31, /* DR_REG_D31 */
+
+    /* from DR_REG_S0 to DR_REG_S31 */
+    DR_REG_V16, /* DR_REG_S0 */
+    DR_REG_V16, /* DR_REG_S1 */
+    DR_REG_V16, /* DR_REG_S2 */
+    DR_REG_V16, /* DR_REG_S3 */
+    DR_REG_V17, /* DR_REG_S4 */
+    DR_REG_V17, /* DR_REG_S5 */
+    DR_REG_V17, /* DR_REG_S6 */
+    DR_REG_V17, /* DR_REG_S7 */
+    DR_REG_V18, /* DR_REG_S8 */
+    DR_REG_V18, /* DR_REG_S9 */
+    DR_REG_V18, /* DR_REG_S10 */
+    DR_REG_V18, /* DR_REG_S11 */
+    DR_REG_V19, /* DR_REG_S12 */
+    DR_REG_V19, /* DR_REG_S13 */
+    DR_REG_V19, /* DR_REG_S14 */
+    DR_REG_V19, /* DR_REG_S15 */
+    DR_REG_V20, /* DR_REG_S16 */
+    DR_REG_V20, /* DR_REG_S17 */
+    DR_REG_V20, /* DR_REG_S18 */
+    DR_REG_V20, /* DR_REG_S19 */
+    DR_REG_V21, /* DR_REG_S20 */
+    DR_REG_V21, /* DR_REG_S21 */
+    DR_REG_V21, /* DR_REG_S22 */
+    DR_REG_V21, /* DR_REG_S23 */
+    DR_REG_V22, /* DR_REG_S24 */
+    DR_REG_V22, /* DR_REG_S25 */
+    DR_REG_V22, /* DR_REG_S26 */
+    DR_REG_V22, /* DR_REG_S27 */
+    DR_REG_V23, /* DR_REG_S28 */
+    DR_REG_V23, /* DR_REG_S29 */
+    DR_REG_V23, /* DR_REG_S30 */
+    DR_REG_V23, /* DR_REG_S31 */
+
+    /* AArch64-only.
```

Comment:
Wait why would aarch64-only things be in the arm/ directory?

Reply:
To be honest, I think I just turned off my brain for a second here and just blindly followed what `dr_reg_fixer[]` does in core/ir/arm/encode.c (https://github.com/DynamoRIO/dynamorio/blob/master/core/ir/arm/encode.c#L287).
I am not even really sure why the motivation behind having those extra register is "but simpler code to not ifdef it" (old comment from a time when arm and aarch64 where together, similar to x86 and x64, perhaps?), as there's nothing to ifdef here; the `dr_reg_fixer[]` (and `d_r_reg_id_to_virtual[]`) indexes for aarch64 registers should just be mapped to DR_REG_INVALID, if I understand this correctly.

Either way, I think both `dr_reg_fixer[]` and `d_r_reg_id_to_virtual[]` should look "the same" (as we are asking future maintainers to keep them "in synch"), so we either keep them as they are or change both? Advice?

Reply:
I guess it is just to avoid ifdefs.  Just leave as is I suppose.